libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -2470,111 +2482,18 @@ string YulUtilFunctions::updateStorageValueFunction(
 
 			auto const& fromStructType = dynamic_cast<StructType const&>(_fromType);
 			auto const& toStructType = dynamic_cast<StructType const&>(_toType);
-			solAssert(fromStructType.structDefinition() == toStructType.structDefinition(), "");
 			solAssert(_offset.value_or(0) == 0, "");
 
 			Whiskers templ(R"(
 				function <functionName>(slot, <?dynamicOffset>offset, </dynamicOffset>value) {
```

Comment:
Does it make sense to combine this template with the template for arrays above? I think almost everyting is the same except for the actual copy funciton that is called.

Reply:
Totally, merged now